src/sys/wait.rs

```diff
@@ -225,6 +227,61 @@ impl WaitStatus {
             WaitStatus::Continued(pid)
         })
     }
+
+    /// Convert a `siginfo_t` as returned by `waitid` to a `WaitStatus`
+    ///
+    /// # Errors
+    ///
+    /// Returns an `Error` corresponding to `EINVAL` for invalid values.
+    ///
+    /// # Safety
+    ///
+    /// siginfo_t is actually a union, not all fields may be initialized.
+    /// The functions si_pid() and si_status() must be valid to call on
+    /// the passed siginfo_t.
+    #[cfg(any(
+        target_os = "android",
+        target_os = "freebsd",
+        target_os = "haiku",
+        target_os = "linux",
+    ))]
+    pub unsafe fn from_siginfo(siginfo: &libc::siginfo_t) -> Result<WaitStatus> {
```

Comment:
Why does this need to be `pub`?

Reply:
I thought it might be useful with sigwaitinfo or sigaction as well, but maybe it would make more sense to create wrappers for these functions as well if somebody is interested in using them. (sigaction's sa_sigaction is not represented well in libc either, as the struct contains a union in C on most platforms...)

Reply:
Yeah, I don't think it's quite useful enough as is.  Please make it private.

Reply:
Done